src/main/java/ru/mercuriev/game/of/life/graph/GraphEngine.java

```diff
@@ -0,0 +1,33 @@
+package ru.mercuriev.game.of.life.graph;
+
+import ru.mercuriev.game.of.life.graph.cell.Cell;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Engine interface implementation
+ * Uses Cell uder the hood to calculate next generation
+ *
+ * @author paul
+ */
+// TODO implement Engine interface
+public class GraphEngine {
+
+    // TODO @Override
+    public int[][] next(int[][] currentGeneration) {
+
+        // int[][] -> Stream<IntStream>
+        Stream<IntStream> input = Arrays.stream(currentGeneration)
+                                        .map(Arrays::stream);
+
+        Stream<IntStream> nextGeneration = Cell.valueOf(input).getNextGeneration();
+
+        // Stream<IntStream> -> int[][]
```

Comment:
Util class for engine may be a good idea, but INHO code snippet like ```Arrays.stream(currentGeneration).map(Arrays::stream);``` does not deserve to become an utility method :)